### ISSUE TITLE
Usage Questions

### ISSUE THREAD

**Issue by Chokerino** [agemagician/ProtTrans#20]
Hello,

I am working on a protein classification problem and tried to use ProtTrans in my project. I have about 500 proteins in the dataset. I first extracted feature embedding and trained ML and DL models on them but was getting worse accuracy than what I had previously achieved. I wanted to try to train BERT like models for classification but I have some doubts regarding the fact that since BERT is trained on English Language data, wouldn't training and classifying sequence data be incorrect? (I converted the sequence into overlapping amino acid windows of length 3 which is similar to an English sentence). I am trying to train my own language model but most models only support 512 while my sequences are way larger.
Is there any way to use Prottrans to do so or is this approach completely incorrect?

**Comment by agemagician** [comment]
Since, you didn't provide a Colab example for your task, and didn't provide more specific details (the specific use-case, previous approach, kind of DL models, etc.), I can only give you a general advice.

1. You should try "[ProtT5-XL-BFD](https://github.com/agemagician/ProtTrans/blob/master/Embedding/PyTorch/Advanced/ProtT5-XL-BFD.ipynb)". This provides better results than "ProtBert-BFD" for classification tasks.

2. If you want to use feature extraction, then you have to perform a lot of experiments in order to find the best model on top of the LM models. For example a single linear layer on top will not provide good results. You should try to add CNN layers or LSTM layers on top.

3. Try different pooling strategies for feature extraction. You can find example for 4 different strategies [here](https://github.com/agemagician/ProtTrans/blob/master/Fine-Tuning/ProtBert-BFD-FineTuning-PyTorchLightning-MS.ipynb).

4. You can try [fine-tuning](https://github.com/agemagician/ProtTrans/tree/master/Fine-Tuning). In this case, you only need to add a single linear layer on top of the model. This approach doesn't require a lot of experiments to reach a good results.

Finally, the approach you mentioned of window of length 3 is an old method used previously on old methods like Word2Vec and Glove, which is not used at all with new LMs like ProtBert and ProtT5. ProtBert support up to 40k sequence length and ProtT5 support unlimited sequence length.
